org.metaborg.spoofax.core/src/main/java/org/metaborg/spoofax/core/stratego/StrategoCommon.java

```diff
@@ -112,6 +155,35 @@
             throw handleException(e, runtime, strategy);
         }
     }
+    
+	@Override
+	public IStrategoTerm invoke(HybridInterpreter runtime, IStrategoTerm input, String strategy,
+			List<IStrategoTerm> termArguments) throws MetaborgException {
+		runtime.setCurrent(input);
+		
+		final IStrategoString strategyName = createStrategyName(strategy, termArguments);
+		final IStrategoAppl strategyNameTerm = termFactory.makeAppl("SVar", strategyName);
+		final IStrategoAppl strategyCallTerm = termFactory.makeAppl("CallT", strategyNameTerm, termFactory.makeList(), termFactory.makeList(termArguments));
+		
+        final IStrategoConstructor sdefT = termFactory.makeConstructor("SDefT", 4);
+		final IStrategoAppl strategyTerm = termFactory.makeAppl(sdefT, strategyName, 
+				termFactory.makeList(), termFactory.makeList(), strategyCallTerm);
```

Comment:
Can you explain what you're doing here? It looks like for calling `my-strategy` with term arguments `1,2,3`, you're creating the following term: `SDefT("my_strategy_0_3", [], [], CallT(SVar("my_strategy_0_3"), [], [1,2,3]))`, which would looks like this in Stratego `my_strategy_0_3(|) = my_stratego_0_3(|1,2,3)`. This makes no sense to me, cified names should already distinguish between strategies with different number of arguments but this one is calling itself with inconsistent numbers of arguments?

Reply:
Tbh, I arrived at this solution through trail and error. Initially I tried to just pass the CallT Term to the evaluate method but that didn't work for some reason. I'll look into it again.

Reply:
Actually it's possible to just pass the CallT term and it works just the same.